packages/material-ui-system/src/styleFunctionSx.test.js

```diff
@@ -127,7 +127,7 @@ describe('styleFunctionSx', () => {
             xl: 0.5,
           },
           border: [1, 2, 3],
-          borderColor: (t) => t.palette.secondary.main,
+          borderColor: (t) => [t.palette.secondary.main, t.palette.primary.main],
```

Comment:
Updated tests for the function changes implemented here https://github.com/mui-org/material-ui/pull/23688/files#r530971112